### PR TITLE
fix: update current http urls as per #960

### PR DESCRIPTION
**Pull request by LexLuthr**
Fixes part of https://github.com/filecoin-project/boost/issues/960

### CODE REVIEW

**Review by dirkmc — COMMENTED** [review]
I suggest we
- remove the `format` parameter from the /piece endpoint, as it can only ever have one value so it's redundant
- remove all code related to building car files
- note that we should also remove any methods from HttpServerApi and its implementation that are no longer used

We can always find this code in the commit history if we need it later

File: cmd/booster-http/server.go [hunk 1]
```diff
@@ -151,122 +124,59 @@ func (s *HttpServer) handleIndex(w http.ResponseWriter, r *http.Request) {
...
+	startTime := time.Now()
+	ctx, span := tracing.Tracer.Start(r.Context(), "http.piece_cid")
+	defer span.End()
+	stats.Record(ctx, metrics.HttpPieceByCidRequestCount.M(1))
+
 	q, err := url.ParseQuery(r.URL.RawQuery)
 	if err != nil {
 		msg := fmt.Sprintf("parsing query: %s", err.Error())
 		writeError(w, r, http.StatusBadRequest, msg)
+		stats.Record(ctx, metrics.HttpPayloadByCid400ResponseCount.M(1))
```
Comment (jacobheun) [inline cmd/booster-http/server.go:146]: I'm not familiar with all of the metrics vars, but as we're only doing piece retrieval by piece cid shouldnt this be `HttpPieceByCid400ResponseCount` or similar?

Same for the below values.

**Comment by LexLuthr** [comment]
@jacobheun Please let me know what's the final call on Dirk's comment. It makes sense to remove that param as it is not adding any value.

**Comment by jacobheun** [comment]
@LexLuthr agreed on removing it as there's only 1 option.

**Comment by dirkmc** [comment]
Let's remove PiecesContainingMultihash and GetMaxPieceOffset from HttpServerApi as they are no longer used.
Let's also remove the implementations from serverApi

**Review by dirkmc — APPROVED** [review]

**Pull request merged**